kubernetes/istio/istio/src/main/java/org/arquillian/cube/istio/impl/IstioAssistant.java

```diff
@@ -0,0 +1,137 @@
+
+public class IstioAssistant {
+
+    private final IstioClient istioClient;
+    private final OkHttpClient httpClient;
+
+    public IstioAssistant(IstioClient istioClient) {
+        this.istioClient = istioClient;
+        this.httpClient = new OkHttpClient();
+    }
+
+    public List<IstioResource> deployIstioResources(final InputStream inputStream) {
+        return istioClient.registerCustomResources(inputStream);
+    }
+
+    public void undeployIstioResources(final List<IstioResource> istioResources) {
+        for (IstioResource istioResource : istioResources) {
+            istioClient.unregisterCustomResource(istioResource);
+        }
+    }
+
+    public List<IstioResource> deployIstioResources(final URL...urls) throws IOException {
+
+        final List<IstioResource> istioResources = new ArrayList<>();
+
+        for (URL url : urls) {
+            try (final InputStream inputStream = url.openStream()) {
+                istioResources.addAll(this.deployIstioResources(inputStream));
+            }
+        }
+
+        return istioResources;
+    }
+
+    /**
+     * Deploys application reading resources from classpath, matching the given regular expression.
+     * For example istio/.*\\.json will deploy all resources ending with json placed at istio classpath directory.
+     *
+     * @param pattern to match the resources.
+     */
+    public List<IstioResource> deployIstioResourcesFromClasspathPattern(String pattern) {
+
+        final List<IstioResource> istioResources = new ArrayList<>();
+        final FastClasspathScanner fastClasspathScanner = new FastClasspathScanner();
+
+        fastClasspathScanner.matchFilenamePattern(pattern, (FileMatchProcessor) (relativePath, inputStream, lengthBytes) -> {
+            istioResources.addAll(deployIstioResources(inputStream));
+            inputStream.close();
+        }).scan();
+
+        return istioResources;
+    }
+
+    /**
+     * Deploys all y(a)ml and json files located at given directory.
+     *
+     * @param directory       where resources files are stored
+     * @throws IOException
+     */
+    public List<IstioResource> deployIstioResources(final Path directory) throws IOException {
+
+        final List<IstioResource> istioResources = new ArrayList<>();
+
+        if (Files.isDirectory(directory)) {
+            Files.list(directory)
+                .filter(ResourceFilter::filterKubernetesResource)
+                .map(p -> {
+                    try {
+                        return Files.newInputStream(p);
+                    } catch (IOException e) {
+                        throw new IllegalArgumentException(e);
+                    }
+                })
+                .forEach(is -> {
+                    try {
+                        istioResources.addAll(deployIstioResources(is));
+                        is.close();
+                    } catch (IOException e) {
+                        throw new IllegalArgumentException(e);
```

Comment:
I am not sure if we are catching `IOException` here, then why throwing it a `IllegalArgumentException`? Maybe `UncheckedIOException` or `RuntimeException` with some kind of msg will help for user